addons/mail/tests/common.py

```diff
@@ -339,19 +339,25 @@ def _find_mail_mail_wrecord(self, record, status=None, mail_message=None, author
             )
         return mail
 
-    def _find_sent_email(self, email_from, emails_to, subject=None):
-        """ Find an outgoing email based on from / to and optional subject when
-        having a conflict.
+    def _find_sent_email(self, email_from, emails_to, subject=None, body=None, attachment_names=None):
+        """ Find an outgoing email based on from / to and optional subject, body
+        and attachment names when having conflicts.
 
         :return sent_email: an outgoing email generated during the mock;
         """
         sent_emails = [
             mail for mail in self._mails
-            if set(mail['email_to']) == set(emails_to) and mail['email_from'] == email_from
+            if (set(email for email in mail['email_to']) == set(email for email in emails_to)
```

Comment:
Does that change anyting ? Seems you are building set from iterators, isn't it the same as before ?